### PR TITLE
feat: Retrieve all job listings 

### PR DESCRIPTION
**Pull request by Asin-Junior-Honore**
### Pull Request: [FEAT] - External Dynamic Pages - Job Listings - Retrieve All Job Listings #145

**Description**

Implemented a feature to retrieve all job listings from the database. This endpoint allows users to fetch job listings with pagination. The response includes job details and pagination information.

**Endpoints**

- `GET /api/v1/jobs`

**Response Format**

```json
{
  "message": "Job listings retrieved successfully.",
  "data": [
    {
      "title": "String",
      "description": "String",
      "location": "String",
      "salary": "String",
      "job_type": "String"
    }
  ],
  "pagination": {
    "current_page": "Int",
    "total_pages": "Int",
    "page_size": "Int",
    "total_items": "Int"
  }
}
```

**Manual Testing**

1. **Retrieve All Job Listings**

   - **Action**: Send a request to the `GET /api/v1/jobs` endpoint.
   - **Expected Result**: You should receive a response with the job listings, including the `message`, `data`, and `pagination` fields.

   **Example Request:**
   ```bash
   curl -X GET http://localhost:3000/api/v1/jobs
   ```

2. **Test Pagination**

   - **Action**: Send a request to the `GET /api/v1/jobs` endpoint with query parameters for pagination (e.g., `page` and `pageSize`).
   - **Expected Result**: The response should include the `pagination` details with `current_page`, `total_pages`, `page_size`, and `total_items`.

   **Example Request:**
   ```bash
   curl -X GET "http://localhost:3000/api/v1/jobs?page=2&pageSize=5"
   ```

**Checklist**

- [x] Implemented an endpoint to retrieve all job listings.
- [x] Added pagination to the job listings endpoint.
- [x] Validated response format includes `message`, `data`, and `pagination`.
- [x] Ensured proper error handling for unexpected issues.

**Reference Issue**

[FEAT] - External Dynamic pages - Job Listings - Retrieve All Job Listings

---

### CODE REVIEW

**Comment by Asin-Junior-Honore** [comment]
@Homoakin619 
@Heba-WebDev  
please review please😢🙏

**Comment by davidolanrewaju** [comment]
Your PR is missing test files, please do well to include them

File: src/modules/jobs/create-job.dto.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { IsString } from 'class-validator';
+
+export class CreateJobDto {
```
Comment (Cobidex) [inline src/modules/jobs/create-job.dto.ts:3]: Add docs

File: src/modules/jobs/job.controller.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { JobService } from './job.service';
+
+@Controller('api/v1/jobs')
+export class JobController {
+  constructor(private readonly jobService: JobService) {}
+
+  @Get()
```
Comment (Cobidex) [inline src/modules/jobs/job.controller.ts:8]: Add docs

File: src/modules/jobs/job.entity.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity()
+export class Job {
```
Comment (Cobidex) [inline src/modules/jobs/job.entity.ts:4]: Extend base class, remove repeated fields: id, created_at and uldated_at

**Review by Cobidex — COMMENTED** [review]
Don't forget test files as pointed out. Then ensure your .env.developmenr.local file is present as directed by Buka

**Comment by Heba-WebDev** [comment]
Please rename your branch. The name is extremely long

**Pull request closed without merging**